src/pages/Profile.tsx

```diff
@@ -389,153 +586,120 @@ class Profile extends Component<Props, ProfileState> {
               const service = services[subscription.name];
               return [
                 ...jsxElementsArray,
-                <div key={subscription.id} className="shadow p-4 my-4">
-                  <SubscriptionService service={service} t={t} />
-                  {this.checkService(service)}
-                  <h6 className="mt-4">
-                    {t("subscription")} ({subscription.state})
-                  </h6>
-                  <div className="my-2">
-                    {t("primary_key")}:{" "}
-                    {this.state.keyDisplay[`p_${subscription.name}`]
-                      ? subscription.primaryKey
-                      : t("key")}
-                    <Button
-                      outline={true}
-                      color="primary"
-                      size="xs"
-                      className="ml-1 mr-1"
-                      onClick={() => this.onToggleKey(`p_${subscription.name}`)}
-                    >
-                      {this.state.keyDisplay[`p_${subscription.name}`] ? (
-                        <FaEyeSlash />
-                      ) : (
-                        <FaEye />
-                      )}
-                    </Button>
-                    <Button
-                      color="danger"
-                      size="xs"
-                      className="mr-1"
-                      disabled={!service || subscription.state !== "active"}
-                      onClick={this.onRegenerateKey(
-                        "primary",
-                        subscription.name
-                      )}
-                    >
-                      {t("regenerate")}
-                    </Button>
-                    <Button
-                      color="primary"
-                      size="xs"
-                      className="mr-1"
-                      disabled={!service || subscription.state !== "active"}
-                      onClick={this.onSetKey(subscription.primaryKey, service)}
-                    >
-                      {t("use")}
-                    </Button>
+                <div className="card-service my-4" key={subscription.id}>
+                  <div className="p-4 mt-4">
+                    <SubscriptionService
+                      service={service}
+                      t={t}
+                      status={this.renderServiceStatus(service)}
+                      atLeastOneVisible={this.state.atLeastOneVisible}
```

Comment:
Do we need a new status value? We can use instead
```suggestion
                      atLeastOneVisible={Object.values(
                        this.state.services
                      ).some(_ => _.is_visible)}
```
What are the purpose of this property? Seems never used

Reply:
Fix with 3819851. The purpose of this prop is to pass to backend a delegate with at least one service in visible status. But we delete this weak feature in a review call.